File: seller/src/main/java/com/example/seller/Adapter/CategoryAdapter.java
package com.example.seller.Adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.navigation.Navigation;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.example.seller.Models.CategoryModel;
import com.example.seller.R;
import com.example.seller.SessionManager;
import com.example.seller.ui.slideshow.SlideshowFragmentDirections;
import com.firebase.ui.database.FirebaseRecyclerAdapter;
import com.firebase.ui.database.FirebaseRecyclerOptions;
import com.google.firebase.database.FirebaseDatabase;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.Objects;


public class CategoryAdapter extends FirebaseRecyclerAdapter<CategoryModel, CategoryAdapter.ViewHolder> {

    boolean isSelected;
    ArrayList<CategoryModel> data;
    Context context;
    public CategoryAdapter(@NonNull @NotNull FirebaseRecyclerOptions<CategoryModel> options) {
        super(options);
    }


    @NonNull
    @NotNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull @NotNull ViewGroup parent, int viewType) {
        context = parent.getContext();
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.sample_menu_category, parent, false);
        return new ViewHolder(view);
    }

    @Override
    protected void onBindViewHolder(@NonNull @NotNull ViewHolder holder, int position, @NonNull @NotNull CategoryModel model) {
        Glide.with(holder.categoryImage.getContext()).load(model.getCategoryImage()).into(holder.categoryImage);
        holder.categoryName.setText(model.getCategoryName());

        holder.itemView.setOnClickListener(v -> {
            String phone = new SessionManager(context).getUserDetailFromSession().get(SessionManager.KEY_PHONE_NO);
            assert phone != null;
            String key = FirebaseDatabase.getInstance().getReference("Seller").child(phone).child("Menu").child(Objects.requireNonNull(getRef(position).getKey())).getKey();
            SlideshowFragmentDirections.ActionNavSlideshowToMenuFragment action =
                    SlideshowFragmentDirections.actionNavSlideshowToMenuFragment(key);
            Navigation.findNavController(v).navigate(action);
        });

        holder.itemView.setOnLongClickListener(new View.OnLongClickListener() {
            @Override
            public boolean onLongClick(View v) {


                return false;

            }
        });
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {

        ImageView categoryImage;
        TextView categoryName;

        public ViewHolder(@NonNull @NotNull View itemView) {
            super(itemView);

            categoryImage = itemView.findViewById(R.id.categoryImage);
            categoryName = itemView.findViewById(R.id.categoryName);
        }
    }
}

File: app/src/main/java/com/example/loginApp/adapters/CategoryAdapter.kt
package com.example.loginApp.adapters

import android.annotation.SuppressLint
import android.content.Context
import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.loginApp.models.CategoryModel
import com.example.loginApp.models.MainModel
import com.example.loginApp.R
import com.firebase.ui.database.FirebaseRecyclerAdapter
import com.firebase.ui.database.FirebaseRecyclerOptions
import com.google.firebase.database.FirebaseDatabase

class CategoryAdapter(options: FirebaseRecyclerOptions<CategoryModel?>, val recyclerView: RecyclerView,
                      private var phoneNo: String, private var allButton : LinearLayout,
                      private var index : Int) : FirebaseRecyclerAdapter<CategoryModel, CategoryAdapter.ViewHolder>(options) {

    private var context: Context? = null

    @SuppressLint("NotifyDataSetChanged")
    override fun onBindViewHolder(holder: ViewHolder, position: Int, model: CategoryModel) {
        Glide.with(holder.categoryImage).load(model.categoryImage).into(holder.categoryImage)
        holder.categoryName.text = model.categoryName

        holder.itemView.setOnClickListener {
            index = position
            notifyDataSetChanged()
            allButton.setBackgroundColor(Color.parseColor("#FFFFFF"))

            val mainOptions = FirebaseRecyclerOptions.Builder<MainModel>().setQuery(
                    FirebaseDatabase.getInstance().reference.child("Seller").child(phoneNo).child("Menu")
                        .orderByChild("key").equalTo(getRef(position).key), MainModel::class.java).build()
            recyclerView.layoutManager = LinearLayoutManager(context)
            val mainAdapter = MainAdapter(mainOptions, phoneNo)
            recyclerView.adapter = mainAdapter
            mainAdapter.startListening()
        }
        if (index == position) {
            holder.linearLayout.background = ContextCompat.getDrawable(context!!, R.drawable.boarder_black)
        } else {
            holder.linearLayout.background = ContextCompat.getDrawable(context!!, R.drawable.background_rv)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        context = parent.context
        val view = LayoutInflater.from(parent.context).inflate(R.layout.sample_category, parent, false)
        return ViewHolder(view)
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var categoryImage : ImageView = itemView.findViewById(R.id.categoryImage)
        var categoryName : TextView = itemView.findViewById(R.id.categoryName)
        val linearLayout : LinearLayout = itemView.findViewById(R.id.linearCategory)
    }
}